cpp/modmesh/mathtype/Complex.hpp

```diff
@@ -0,0 +1,116 @@
+ *   and/or other materials provided with the distribution.
+ * - Neither the name of the copyright holder nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <type_traits>
+#include <cmath>
+
+namespace modmesh
+{
+
+namespace detail
+{
+template <typename T, typename = std::enable_if_t<std::is_arithmetic<T>::value>>
+struct ComplexImpl
+{
+    T real_v;
+    T imag_v;
+
+    explicit ComplexImpl()
+        : ComplexImpl(0.0, 0.0)
+    {
+    }
+
+    explicit ComplexImpl(T r, T i)
+        : real_v(r)
+        , imag_v(i)
+    {
+    }
+
+    explicit ComplexImpl(const ComplexImpl & other)
+        : real_v(other.real_v)
+        , imag_v(other.imag_v)
+    {
+    }
+
+    ComplexImpl operator+(const ComplexImpl & other) const
+    {
+        return ComplexImpl(real_v + other.real_v, imag_v + other.imag_v);
+    }
+
+    ComplexImpl operator-(const ComplexImpl & other) const
+    {
+        return ComplexImpl(real_v - other.real_v, imag_v - other.imag_v);
+    }
+
+    ComplexImpl operator*(const ComplexImpl & other) const
+    {
+        return ComplexImpl(real_v * other.real_v - imag_v * other.imag_v, real_v * other.imag_v + imag_v * other.real_v);
+    }
+
+    ComplexImpl operator/(const T & other) const
+    {
+        return ComplexImpl(real_v / other, imag_v / other);
+    }
+
+    ComplexImpl & operator=(const ComplexImpl & other)
+    {
+        if (this != &other) // Check for self-assignment
+        {
+            real_v = other.real_v;
+            imag_v = other.imag_v;
+        }
+        return *this;
+    }
+
+    ComplexImpl & operator+=(const ComplexImpl & other)
+    {
+        real_v += other.real_v;
+        imag_v += other.imag_v;
+        return *this;
+    }
+
+    ComplexImpl & operator-=(const ComplexImpl & other)
+    {
+        real_v -= other.real_v;
+        imag_v -= other.imag_v;
+        return *this;
+    }
+
+    T real() const { return real_v; }
+    T imag() const { return imag_v; }
+    T norm() const { return real_v * real_v + imag_v * imag_v; }
+}; /* end class ComplexImpl */
```

Comment:
Meanwhile, I'm trying to integrate with SimpleArray and create an array type Python wrapper and make it compatible with the NumPy buffer protocol, but I got a static assert https://github.com/solvcon/modmesh/blob/c9594da86f25b12dcacb68444b1badf8f181b1b0/cpp/modmesh/python/common.hpp#L63
Which is that `ComplexImpl<T>` is not a POD type. Currently, I don't have a firm plan to move forward.

Reply:
The ending mark should be `/* end structure ComplexImpl */`.

Reply:
`ComplexImpl<T>` looks like POD.  What does it mean to say t is not a POD type?